src/routes/home.tsx

```diff
@@ -5,27 +5,55 @@ import Spinner from '@/components/spinner'
 import { compareEvents } from '@/utils/compare-events'
 import { useGeoLocation } from '@/utils/use-geo-location'
 import { useEvents } from '@/cache/events/use'
+import { useState } from 'preact/hooks'
 
 const now = new Date()
-
 const Home = () => {
   const events = useEvents()
   const location = useGeoLocation()
+  const [term = '', setTerm] = useState(0);
+
+  function searchingFor(term: string | number) {
+    return function (x: any) {
+      // because not all objects contain properties 'district' and 'fullDistrict' you have to check if particular object contains property 'distict' and if not tell method to not use it. Otherwise this error will occur (Uncaught (in promise) TypeError: Cannot read property 'toString' of undefined)
+      if (x.hasOwnProperty('district')) {
+        return (x.name.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
+          (x.key.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
+          (x.locationName.toString().toLowerCase().includes(term.toString().toLowerCase()) || !term) ||
```

Comment:
`event.key` and `event.locationName` exist regardless of whether the `district` property exists, so I don't think they should be within this `if` statement that checks whether `district` exists.

Reply:
But I think that, then you will not be able to filter events correctly. For example : 
You have some event which has "property: district" and you are trying to find this event according his "property:key" .
So you type key into search bar ,but this event will not be displayed beacuse you are not checking if event.key match with what you typed into search bar. You checking it only in case if object has not property "district".
Or am I wrong? I also tried to remove it from if statement in code and it doesnt work then.

Reply:
You are right. I think there is a different way of handling this that allows us to not duplicate these conditions. I'll suggest it after you apply the other suggestions.